### PR TITLE
Release Python3.8 wheels for Windows

### PR DESCRIPTION
**Pull request by lidizheng**
Related: https://github.com/grpc/grpc/issues/20615
This PR pre-installs Python 3.8 on Kokoro Windows (VM?). It should enable us to release 3.8 for Windows users.

Let's see.

### CODE REVIEW

**Comment by lidizheng** [comment]
[prod:grpc/core/experimental/grpc_build_artifacts_multiplatform](http://sponge/46ab648b-9378-42f7-8045-14b8e0a8b8d4)

**Comment by gnossen** [comment]
Looks like a TLS failure?
```
T:\src\github\grpc>powershell -command "(New-Object Net.WebClient).DownloadFile('https://www.python.org/ftp/python/3.8.0/python-3.8.0-amd64.exe', 'C:\tools\python-3.8.0-amd64.exe')" 
Exception calling "DownloadFile" with "2" argument(s): "The request was 
aborted: Could not create SSL/TLS secure channel."
At line:1 char:1
+ (New-Object 
Net.WebClient).DownloadFile('https://www.python.org/ftp/python/3.8.0 ...'
```

File: tools/internal_ci/helper_scripts/prepare_build_windows.bat [hunk 1]
```diff
@@ -35,6 +35,14 @@ python -m pip install google-api-python-client
 powershell -File src\csharp\install_dotnet_sdk.ps1
 set PATH=%LOCALAPPDATA%\Microsoft\dotnet;%PATH%
 
+@rem Install Python 3.8.0
+powershell -command "(New-Object Net.WebClient).DownloadFile('https://www.python.org/ftp/python/3.8.0/python-3.8.0-amd64.exe', 'C:\tools\python-3.8.0-amd64.exe')"
+powershell -command "& 'C:\tools\python-3.8.0-amd64.exe' /quiet InstallAllUsers=1 PrependPath=1 Include_test=0 TargetDir='C:\Python38'"
```
Comment (gnossen) [inline tools/internal_ci/helper_scripts/prepare_build_windows.bat]: For security purposes, we'll want to [verify these downloads with a checksum](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/get-filehash?view=powershell-6).

**Comment by lidizheng** [comment]
[prod:grpc/core/experimental/grpc_build_artifacts_multiplatform](http://sponge/1399e3c9-8928-422f-b5a1-cd6f8f908bc4)

File: tools/internal_ci/helper_scripts/prepare_build_windows.bat (hunk 1 above)
Comment (lidizheng) [reply tools/internal_ci/helper_scripts/prepare_build_windows.bat]: Done in the script. Let's see if Kokoro is happy this time.

**Comment by lidizheng** [comment]
[prod:grpc/core/experimental/grpc_build_artifacts_multiplatform](http://sponge/a112278e-f0f4-4d5c-9eda-33a36573bf65)

File: tools/internal_ci/windows/grpc_build_artifacts.bat [hunk 2]
```diff
@@ -12,13 +12,6 @@
 @rem See the License for the specific language governing permissions and
 @rem limitations under the License.
 
-@rem Move python installation from _32bit to _32bits where they are expected by python artifact builder
```
Comment (gnossen) [inline tools/internal_ci/windows/grpc_build_artifacts.bat]: :heart: :heart: :heart:

File: tools/internal_ci/helper_scripts/install_python38.ps1 [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    Write-Host "Python installer $PythonInstallerPath validated."
+
+    # Installs Python
+    & $PythonInstallerPath /passive InstallAllUsers=1 PrependPath=1 Include_test=0 TargetDir=$PythonInstallPath
+    if (-Not $?) {
+        throw "The Python installation exited with error!"
+    }
+
+    # Validates Python
```
[the remark this answers is not in the document]
Comment (lidizheng) [reply tools/internal_ci/helper_scripts/install_python38.ps1]: Longer comment added.

File: tools/internal_ci/helper_scripts/install_python38.ps1 [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    $PythonBinary = "$PythonInstallPath\python.exe"
+    while ($true) {
```
[the remark this answers is not in the document]
Comment (lidizheng) [reply tools/internal_ci/helper_scripts/install_python38.ps1]: The 5 minutes deadline added.

I was concerning that even if we poison this script, we can't stop the execution of the rest of the CI build. Admittedly, adding deadline is better.

**Comment by lidizheng** [comment]
[prod:grpc/core/experimental/grpc_build_artifacts_multiplatform](http://sponge/a734a5fc-a005-412b-bbfb-4e6abb2e7d5d)

**Comment by lidizheng** [comment]
Known failures: https://github.com/grpc/grpc/issues/21320 https://github.com/grpc/grpc/issues/20020

**Pull request merged**